pkg/util/timeutil/pgdate/parsing.go

```diff
@@ -65,9 +64,21 @@ var (
 // These are sentinel values for handling special values:
 // https://www.postgresql.org/docs/10/static/datatype-datetime.html#DATATYPE-DATETIME-SPECIAL-TABLE
 var (
-	TimeEpoch            = timeutil.Unix(0, 0)
-	TimeInfinity         = timeutil.Unix(math.MaxInt64, math.MaxInt64)
-	TimeNegativeInfinity = timeutil.Unix(math.MinInt64, math.MinInt64)
+	TimeEpoch = timeutil.Unix(0, 0)
+	// TimeInfinity represents the "highest" possible time.
+	// TODO (#41564): this should actually behave as infinity, i.e. any operator
+	// leaves this as infinity. This time should always be greater than any other time.
+	// We should probably use the next microsecond after this value, i.e. timeutil.Unix(9224318016000, 0).
+	// Postgres uses math.MaxInt64 microseconds as the infinity value.
+	// See: https://github.com/postgres/postgres/blob/42aa1f0ab321fd43cbfdd875dd9e13940b485900/src/include/datatype/timestamp.h#L107.
+	TimeInfinity = timeutil.Unix(9224318016000-1, 999999000)
```

Comment:
Just to double check, we were already returning these when seeing "infinity", but they just overflowed here?

Reply:
we were returning a value that overflowed to negative date value